numpy/random/mtrand/mtrand.pyx

```diff
@@ -4143,15 +4143,15 @@ cdef class RandomState:
         if op.shape == ():
             fp = PyFloat_AsDouble(p)
 
-            if fp < 0.0:
-                raise ValueError("p < 0.0")
+            if fp <= 0.0:
+                raise ValueError("p <= 0.0")
```

Comment:
It might be suitable to add a unit test for both scalar and array `p` values of 0 (that failed before and pass now)--ensuring that `ValueError` is appropriately raised.